pegin/getaddress.go

```diff
@@ -0,0 +1,142 @@
+package pegin
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+
+	"github.com/btcsuite/btcutil"
+
+	"github.com/btcsuite/btcd/txscript"
+
+	"github.com/vulpemventures/go-elements/network"
+	"github.com/vulpemventures/go-elements/payment"
+
+	"github.com/btcsuite/btcd/chaincfg"
+
+	"github.com/vulpemventures/go-elements/address"
+
+	"github.com/btcsuite/btcd/btcec"
+)
+
+const (
+	MainNet NetworkType = iota
+	RegtestNet
+)
+
+type NetworkType int
+
+type AddressInfo struct {
+	ClaimScript      string
+	MainChainAddress string
+}
+
+// GetAddressInfo simulates getpeginaddress RPC call which returns mainChain
+// address to which BTC should be sent in BTC network and claim script that is
+// to be used in claimpegin RPC alongside with bitcoinTx and txoutproof
+func GetAddressInfo(
+	privateKey btcec.PrivateKey,
+	fedpegInfo FedpegInfo,
+	networkType NetworkType,
+	isDynaFedEnabled bool,
+	contract []byte,
+) (*AddressInfo, error) {
+	liquidNetwork, btcNetwork := getNetworkParams(networkType)
+
+	p2wkh := getClaimWitnessScript(privateKey, liquidNetwork)
+
+	mainChainAddress, err := createMainChainAddress(
+		contract,
+		btcNetwork,
+		isDynaFedEnabled,
+		fedpegInfo.FedpegScript,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return &AddressInfo{
+		ClaimScript:      p2wkh,
+		MainChainAddress: mainChainAddress,
+	}, nil
+}
+
+func getNetworkParams(
+	networkType NetworkType,
+) (*network.Network, *chaincfg.Params) {
+	var liquidNetwork *network.Network
+	var btcNetwork *chaincfg.Params
+	switch networkType {
+	case MainNet:
+		liquidNetwork = &network.Liquid
+		btcNetwork = &chaincfg.MainNetParams
+	case RegtestNet:
+		liquidNetwork = &network.Regtest
+		btcNetwork = &chaincfg.RegressionNetParams
+	}
+
+	return liquidNetwork, btcNetwork
+}
+
+func getClaimWitnessScript(
+	privateKey btcec.PrivateKey,
+	net *network.Network,
+) string {
+	p2wpkh := payment.FromPublicKey(
+		privateKey.PubKey(),
```

Comment:
no need to ask for private key secret material, just the public seems ok to me to generate the claiming script

Reply:
Done